src/azul/terra.py

```diff
@@ -405,38 +407,42 @@ def _insufficient_access(self, resource: str) -> Exception:
         return self.credentials_provider.insufficient_access(resource)
 
 
+@attr.s(frozen=True, kw_only=True, auto_attribs=True)
+class TDRSnapshot:
+    # FIXME TDR's enumerateSnapshots response lacks name of Google project
+    #       https://github.com/HumanCellAtlas/dcp2/issues/54
+    project: Optional[str]
+    id: str
+    name: str
+    location: str
+
+
 class TDRClient(SAMClient):
     """
     A client for the Broad Institute's Terra Data Repository aka "Jade".
     """
 
-    @attr.s(frozen=True, kw_only=True, auto_attribs=True)
-    class TDRSource:
-        project: str
-        id: str
-        location: str
-
     @cache
-    def lookup_source(self, source_spec: TDRSourceSpec) -> TDRSource:
-        source = self._lookup_source(source_spec)
-        storage = one(
-            storage
-            for dataset in (s['dataset'] for s in source['source'])
-            for storage in dataset['storage']
-            if storage['cloudResource'] == 'bigquery'
-        )
-        return self.TDRSource(project=source['dataProject'],
-                              id=source['id'],
-                              location=storage['region'])
+    def get_snapshot(self, source_spec: TDRSourceSpec) -> TDRSnapshot:
+        snapshot = self._get_snapshot(source_spec)
+        id = snapshot['id']
+        name = snapshot['name']
+        require(source_spec.name == name,
+                f'Sources {source_spec.name!r} and {name!r} both have ID {id}')
```

Comment:
A little confused by this comment, how do we know that the `id` is the same?

Reply:
I made myself very confused while trying to figure out whether this message was accurate or not and decided that either way it was just poorly worded.